packages/sessions-sdk-react/src/components/modal-dialog.tsx

```diff
@@ -1,31 +1,49 @@
-import type { ComponentProps, ReactNode } from "react";
-import { Dialog, Modal, ModalOverlay, Heading } from "react-aria-components";
+import clsx from "clsx";
+import { AnimatePresence, motion } from "motion/react";
+import type { ComponentProps } from "react";
+import { Dialog, Modal, ModalOverlay } from "react-aria-components";
 
 import styles from "./modal-dialog.module.css";
 
-type Props = Omit<ComponentProps<typeof ModalOverlay>, "children"> & {
+const MotionModal = motion.create(Modal);
+const MotionModalOverlay = motion.create(ModalOverlay);
+
+type Props = Omit<ComponentProps<typeof MotionModalOverlay>, "children"> & {
   children: ComponentProps<typeof Modal>["children"];
-  heading: ReactNode;
-  message?: ReactNode | undefined;
+  dialogClassName?: string | undefined;
 };
 
 export const ModalDialog = ({
   children,
-  heading,
-  message,
+  isOpen,
+  dialogClassName,
   ...props
 }: Props) => (
-  <ModalOverlay isDismissable className={styles.modalOverlay ?? ""} {...props}>
-    <Modal isDismissable className={styles.modal ?? ""}>
-      {(args) => (
-        <Dialog className={styles.dialog ?? ""}>
-          <Heading slot="title" className={styles.heading ?? ""}>
-            {heading}
-          </Heading>
-          {message && <div className={styles.message}>{message}</div>}
-          {typeof children === "function" ? children(args) : children}
-        </Dialog>
-      )}
-    </Modal>
-  </ModalOverlay>
+  <AnimatePresence>
```

Comment:
similar comment here about being rather generic but potentially useful to have for other projects